test/server/asqJavaqPlugin.test.js

```diff
@@ -0,0 +1,312 @@
+      $(el).attr('uid').should.not.equal("a-uid");
+
+      //this already has one
+      el = $("#uid")[0];
+      this.asqJavaq.processEl($, el);
+      $(el).attr('uid').should.exist;
+      $(el).attr('uid').should.equal("a-uid");
+    });
+
+
+    it("should find the stem if it exists", function(done){
+      var $ = cheerio.load(this.simpleHtml);
+      var el = $(this.tagName)[0];
+      var elWithHtmlInStem = $(this.tagName)[1];
+
+      var result = this.asqJavaq.processEl($, el);
+      result.then(function(result){
+        expect(result.data.stem).to.equal("This is a stem");
+      }.bind(this))
+      .catch(function(err){
+        done(err);
+      });
+
+      var result = this.asqJavaq.processEl($, elWithHtmlInStem);
+      result.then(function(result){
+        expect(result.data.stem).to.equal("This is a stem <em>with some HTML</em>");
+      }.bind(this))
+      .catch(function(err){
+        done(err);
+      });
+
+      var $ = cheerio.load(this.noStemHtml);
+      var el = $(this.tagName)[0];
+      var result = this.asqJavaq.processEl($, el);
+      result.then(function(result){
+        expect(result.data.stem).to.equal("");
+        done();
+      }.bind(this))
+      .catch(function(err){
+        done(err);
+      });
+    });
+
+    it("should return correct data", function(done){
+      var $ = cheerio.load(this.simpleHtml);
+      var el = $(this.tagName)[1];
+
+      var result = this.asqJavaq.processEl($, el);
+      result.then(function(result){
+        expect(result._id).to.equal("a-uid");
+        expect(result.type).to.equal(this.tagName);
+        expect(result.data.stem).to.equal("This is a stem <em>with some HTML</em>");
+
+        done();
+      }.bind(this))
+      .catch(function(err){
+        done(err);
+      });
+    });
+  });
+
+  describe("parseSettings", function(){
+    const defaultQuestionSettings = [
+      {
+        "key": "compileTimeoutMs",
+        "value": "10000",
+        "kind": "number",
+        "level": "question"
+      },
+      {
+        "key": "executionTimeoutMs",
+        "value": "2000",
+        "kind": "number",
+        "level": "question"
+      },
+      {
+        "key": "spamTimeoutMs",
+        "value": "1000",
+        "kind": "number",
+        "level": "question"
+      },
+      {
+        "key": "charactersMaxLength",
+        "value": "10000",
+        "kind": "number",
+        "level": "question"
+      },
+      {
+        "key": "exerciseName",
+        "value": "none",
+        "kind": "string",
+        "level": "question"
+      },
+    ]
+
+    before(function(){
+     // sinon.stub(this.asq.prototype, "parseSettings").returns(Promise.resolve({}));
+    });
+
+    beforeEach(function(){
+      this.asqJavaq = new this.asqJavaqPlugin(this.asq);
+    });
+
+    it("default settings should be persisted to the db with no attributes for initial settings", function(done){
+      var $ = cheerio.load(this.simpleHtml);
+      var el = $(this.tagName)[0];
```

Comment:
give meaningful id's to the elements instead of getting them with array indices. And then do `var el = $('#mcNoInitialSettings').get(0)`